validation-settings/src/ValidationSettings.liquid

```diff
@@ -8,98 +9,1076 @@ import {
   Section,
   NumberField,
   BlockStack,
+  Banner,
+  InlineStack,
+  Image,
 } from "@shopify/ui-extensions-react/admin";
 
-// The target used here must match the target used in the extension's toml file (../shopify.extension.toml)
 const TARGET = "admin.settings.validation.render";
 
-export default reactExtension(TARGET, () => <ValidationSettings />);
+export default reactExtension(TARGET, async (api) => {
+  const configuration = JSON.parse(
+    api.data.validation?.metafields?.[0]?.value ?? "{}",
+  );
+
+  if (!api.data.validation?.metafields) {
```

Comment:
could this be inferred by the `const configruation` up above?

Reply:
I extracted shared data access to a var 😄